pkg/vectorize/cosh/cosh.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cosh
+
+import "math"
+
+var (
+	coshUint8   func([]uint8, []float64) []float64
+	coshUint16  func([]uint16, []float64) []float64
+	coshUint32  func([]uint32, []float64) []float64
+	coshUint64  func([]uint64, []float64) []float64
+	coshInt8    func([]int8, []float64) []float64
+	coshInt16   func([]int16, []float64) []float64
+	coshInt32   func([]int32, []float64) []float64
+	coshInt64   func([]int64, []float64) []float64
+	coshFloat32 func([]float32, []float64) []float64
+	coshFloat64 func([]float64, []float64) []float64
+)
```

Comment:
perhaps it would be better if you follow the updated function variable initiation in pr#2225

Reply:
So should "coshInt64", "coshFloat32", "coshFloat64" instead be "coshint64", "coshfloat32", "coshfloat64"

Reply:
I think our function variable initiation now is like this
```go
var (
    CoshUint8
)

func init() {
    CoshUint8 = coshUint8
}
```

and Golang supports generic programming now, you can try using it in your cosh implementation if interested